libs/NfcCoreLib/lib/LibNfc/phLibNfc_Init.c

```diff
@@ -106,30 +106,35 @@ static NFCSTATUS phLibNfc_InitCb(void* pContext,NFCSTATUS wStatus,void* pInfo)
-                        if (pHciContext == NULL)
+                        /*The Static HCI Connection exists after NFCC initialization without needing to be
+                        *created using the connection Control Messages defined in Section 4.4.2 and is never closed
+                        */
+                        if (pLibContext->pHciContext == NULL &&
+                            pNciContext->InitRspParams.DataHCIPktPayloadLen > 0)
                         {
-                            /* Failed to allocate memory for HCI packet */
-                            wStatus = NFCSTATUS_FAILED;
-                        }
-                        else
-                        {
-                            pLibContext->pHciContext = pHciContext;
-                            phOsalNfc_SetMemory(pHciContext, 0, sizeof(phHciNfc_HciContext_t));
-                            pHciContext->pNciContext = pNciContext;
-                            pHciContext->bLogDataMessages = !!pLibContext->Config.bLogNciDataMessages;
-
-                            pLibContext->tSeInfo.bSeState[phLibNfc_SE_Index_HciNwk] = phLibNfc_SeStateInitializing;
-                            pLibContext->sSeContext.pActiveSeInfo = (pphLibNfc_SE_List_t)(&pLibContext->tSeInfo.tSeList[phLibNfc_SE_Index_HciNwk]);
-                            wStatus = phNciNfc_UpdateConnDestInfo(UNASSIGNED_DESTID, phNciNfc_e_NFCEE, NULL);
+                            pHciContext = (phHciNfc_HciContext_t*)phOsalNfc_GetMemory(sizeof(phHciNfc_HciContext_t));
+                            if (pHciContext == NULL)
+                            {
+                                /* Failed to allocate memory for HCI packet */
+                                wStatus = NFCSTATUS_FAILED;
+                            }
+                            else
+                            {
+                                pLibContext->pHciContext = pHciContext;
+                                phOsalNfc_SetMemory(pHciContext, 0, sizeof(phHciNfc_HciContext_t));
+                                pHciContext->pNciContext = pNciContext;
+                                pHciContext->bLogDataMessages = !!pLibContext->Config.bLogNciDataMessages;
+
+                                pLibContext->tSeInfo.bSeState[phLibNfc_SE_Index_HciNwk] = phLibNfc_SeStateInitializing;
+                                pLibContext->sSeContext.pActiveSeInfo = (pphLibNfc_SE_List_t)(&pLibContext->tSeInfo.tSeList[phLibNfc_SE_Index_HciNwk]);
+                                wStatus = phNciNfc_UpdateConnDestInfo(UNASSIGNED_DESTID, phNciNfc_e_NFCEE, NULL);
+                            }
                         }
+
+                        /* MaxNFCVFrame size - 4 bytes (1 byte RES_FLAG + 1 byte Parameter + 2 bytes CRC */
+                        pLibContext->ndef_cntx.psNdefMap->ISO15693Container.max_frame_size = pNciContext->InitRspParams.MaxNFCVFrameSize - 4;
```

Comment:
Is there a guarantee that all of the chained memory references in `pLibContext->ndef_cntx.psNdefMap->ISO15693Container` are valid at the time of this call?
Since MaxNFCVFrameSize is specific to the NFCC and doesn't change, it can be stored either in `phFriNfc_NdefMap` or even `phLibNfc_NdefInfo`

Reply:
The changes addressing this have been pushed to the PR branch, closing.